src/Uno.Extensions.Reactive.UI/Presentation/Bindings/BindableListFeed.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Uno.Extensions.Reactive.Bindings.Collections;
+using Uno.Extensions.Reactive.Core;
+
+namespace Uno.Extensions.Reactive.Bindings;
+
+/// <summary>
+/// An helper class use to data-bind a <see cref="IListFeed{T}"/>.
+/// </summary>
+/// <typeparam name="T">The type of the items.</typeparam>
+public sealed partial class BindableListFeed<T> : ISignal<IMessage>, IListInput<T>, IInput<IImmutableList<T>>
+{
+	private readonly CancellationTokenSource _ct = new();
+	private readonly BindableCollection _items;
+	private readonly IState<IImmutableList<T>> _state;
+
+	/// <summary>
+	/// Creates a new instance.
+	/// </summary>
+	/// <param name="propertyName">The name of the property backed by the object.</param>
+	/// <param name="source">The source data stream.</param>
+	/// <param name="ctx">The context of the owner.</param>
+	public BindableListFeed(string propertyName, IListFeed<T> source, SourceContext ctx)
+	{
+		PropertyName = propertyName;
+		_items = BindableCollection.Create<T>();
+		_state = ctx.GetOrCreateState(source.AsFeed());
+
+		((StateImpl<IImmutableList<T>>)_state).GetSource(_ct.Token).ForEachAsync(
+			msg => _items.Switch(new ImmutableObservableCollection<T>(msg.Current.Data.SomeOrDefault(ImmutableList<T>.Empty)!)),
```

Comment:
Why do you need the `!` ?

Reply:
Because the null annotations are wrong on `Option` and I don't want to changed it in that PR. I've already fixed in my next branch, will also remove that `!`